### DISCUSSION TITLE
Netbox is loading too slow /sometimes throwing 504 error and csrf token issue

### DISCUSSION THREAD

**Discussion by jatin9694** [netbox-community/netbox#15382] [Help Wanted!]
### Deployment Type

Self-hosted

### NetBox Version

v3.73

### Python Version

3.9

### Steps to Reproduce

1. I have few problems in netbox while loading it is giving sometimes 504 gateway erorr.
2. sometimes it is loading too fast and while logout giving error.
3. if i click login two three times it is throwing me the csrf token error.

### Expected Behavior

it should load smoothly and quickly.

errors should not be there 

### Observed Behavior

Hello Colleagues,

1. I have few problems in netbox while loading it is giving sometimes 504 gateway erorr.
2. sometimes it is loading too fast and while logout giving error.
3. if i click login two three times it is throwing me the csrf token error.

nginx configuration is fine 

root@NC-NETBOX-P01:/opt/netbox/netbox/netbox# nginx -t
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf 

<img width="1493" alt="Screenshot 2024-03-08 at 00 37 34" src="https://github.com/netbox-community/netbox/assets/102926575/8e8bb63c-1de0-4740-8886-23d3dea5b1ca">

kindly help me in resolving this issue.

Regards,
Jatinder kumar

**Comment by jatin9694** [comment]
Hello Colleagues,

Any updates.

Regards,
Jatinder kumar

**Reply by candlerb** [reply]
The reason you've not got any reply is because nobody else is seeing the behaviour that you see, so it must be something about your system - but you've not said anything about how you're running Netbox that might give any clues. e.g. regular install or docker-compose/kubernetes? Physical hardware, own VM server, cloud VM - which provider? Any sort of load balancer, API gateway or CDN in front? What sort of login authentication have you configured? Are your clients behind any sort of proxy?  Nor have you shown any logs or any other information that might be pertinent, or any configs (e.g. nginx config).

Intermittent CSRF errors suggest proxying, caching or load-balancing issues.

You'll need to check your logs - Netbox/gunicorn logs, nginx logs, and logs from any upstream proxies or caches, to determine what's going on for the failed requests.
